docs/user-guide/configuring.md

```diff
@@ -664,6 +664,7 @@ In each case, the settings in the configuration file override default settings.
 ESLint supports configuration files in several formats:
 
 * **JavaScript** - use `.eslintrc.js` and export an object containing your configuration.
+* **JavaScript (ESM)** - same as `.eslintrc.js` but used with ES module packages
```

Comment:
if `.eslintrc.js` as ESM can be supported, then should `.eslintrc.mjs` be added to the list as well?

Reply:
I had the same thought

Currently, ESLint only works with CommonJS-based configs; rendering the need for a `.mjs` extension unnecessary. If support for ESM-based configurations is added in the future then the addition of `.mjs` extension will likely be necessary.

This PR and the preceding RFC was intentionally made with the following assumptions:
- ESLint itself is written in CommonJS
- Javascript configuration files for ESLint will be written in CommonJS format

Reply:
In that case, this docs line seems misleading, since eslint does *not* support "JavaScript (ESM)"-based configuration?

Reply:
it would be good to see mjs here too

Reply:
@TrejGun As discussed above, this is currently not possible. Please follow along on [this RFC](https://github.com/eslint/rfcs/pull/40) to see discussion around how ESLint is trying to move towards being able to support ESM configuration files and plugins.